.azure-pipelines/common-templates/download-openapi-docs.yml

```diff
@@ -73,12 +73,16 @@ steps:
           $diff = git diff --name-only
           $ModulesWithChanges = @{}
           $diff | %{
-            if (($_ -match 'openApiDocs\/(v1.0|beta)\/(.*).yml') -and !$ModulesWithChanges.ContainsKey($matches.2))
-            {
-              $ModulesWithChanges.Add($matches.2, $matches.1)
+            if ($_ -match 'openApiDocs\/(v1.0|beta)\/(.*)\.yml') {
+              $version = if ($matches[1] -eq 'v1.0') { 'v1.0' } else { 'beta' }
+              $moduleName = "$($matches[2])_$version"
+              if (!$ModulesWithChanges.ContainsKey($moduleName)) {
+                $ModulesWithChanges.Add($moduleName, $matches[1])
+              }
```

Comment:
Seeing a script this complex inline in a pipeline makes me a bit nervous.
yaml is not a great authoring tool for multiline scripts. 
If this were refactored into a stand-alone script that you could call then your can test it independent of a pipeline run.
Plus you could eliminate the if statment in here an use a condition on the step combined with setting the ModulesWithChanges variable to 'Skipped' by default.

Reply:
Gotcha. Included the above runs as references showing that the scripts did work. Initially I was just editing the inline scripts that were already there, however as the complexity grew they definitely should have been moved to their own file. Will update with the changes.